src/ol/renderer/canvas/ImageLayer.js

```diff
@@ -211,7 +227,44 @@ class CanvasImageLayerRenderer extends CanvasLayerRenderer {
       }
     }
 
-    const img = this.getImage();
+    /**
+     * @type {HTMLImageElement|HTMLCanvasElement|HTMLVideoElement|ImageBitmap}
+     */
+    let img = image.getImage();
+    if (img !== this.previousImg_ && this.imageBitmap_) {
+      this.imageBitmap_ = null;
+    }
+    if (
+      !this.getLayer().getSource()?.scaleable &&
+      !(img instanceof HTMLCanvasElement) &&
+      img === this.previousImg_
+    ) {
+      if (!this.imageBitmap_) {
+        // First draw image to a canvas to convert the image data to the fastest format possible
+        // for repeatedly calling drawImage(), then store the result in an image bitmap to avoid
+        // canvas memory shortage in Safari.
+        const context = this.conversionContext_;
+        context.canvas.width = img.width;
+        context.canvas.height = img.height;
+        context.drawImage(img, 0, 0);
+        createImageBitmap(context.canvas)
```

Comment:
Can `createImageBitmap` not use `img` directly?

Reply:
Yes it can, but as the comment says, for some reason the ImageBitmap resulting from a canvas performs way better.

Reply:
OK, if the temporary use of a large canvas is not a problem then that would only be an issue with very large non-SVG images.  if the canvas dimensions exceed these values https://developer.mozilla.org/en-US/docs/Web/HTML/Element/canvas#maximum_canvas_size nothing is drawn beyond that point, and if the area exceeds the maximum the `drawImage` fails silently (at least on Chrome).

Reply:
Yeah, good point, I added a commit which safeguards against that.